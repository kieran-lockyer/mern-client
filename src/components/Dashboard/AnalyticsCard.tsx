import React, { Component } from "react";
import AnalyticsItem from "./AnalyticsItem";
import styled from "styled-components";
import { connect } from "react-redux";

class AnalyticsCard extends Component<any, any> {
  renderPopTags = () => {
    return this.props.popular.map(item => {
      return <AnalyticsItem tag={item._id} count={item.count} key={item._id} />;
    });
  };

  renderTrendingTags = () => {
    return this.props.trending.map(item => {
      return <AnalyticsItem tag={item._id} count={item.count} key={item._id} />;
    });
  };

  renderCard = () => {
    return this.props.title === "Most Popular Tags"
      ? this.renderPopTags()
      : this.renderTrendingTags();
  };

  render() {
    return (
      <Card>
        <CardHeading>
          <h3>{this.props.title}</h3>
        </CardHeading>
        {this.renderCard()}
      </Card>
    );
  }
}

const Card = styled.div`
  background: #fff;
  flex: 1 1 48%;
  margin: 10px 10px 30px 0;
  border-radius: 15px;
  align-self: start;
  &:nth-child(2n),
  &:last-child {
    margin-right: 0;
  }

  @media only screen and (max-width: 1000px) {
    flex-basis: 100%;
    margin: 0 0 1rem 0;
  }
`;

const CardHeading = styled.div`
  color: #fff;
  background: linear-gradient(to right bottom, #223f67, #192b44);
  border-top-left-radius: inherit;
  border-top-right-radius: inherit;
  border-bottom: 1px solid #eee;
  padding: 1rem 2rem;
  display: flex;
  align-items: center;

  h3 {
    margin: 0;
  }
`;

const mapStateToProps = state => ({
  popular: state.stats.popTags,
  trending: state.stats.trendingTags
});

export default connect(mapStateToProps)(AnalyticsCard);
