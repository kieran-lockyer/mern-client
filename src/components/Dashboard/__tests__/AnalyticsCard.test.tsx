import React from 'react';
import { shallow } from 'enzyme';
import AnalyticsCard from '../AnalyticsCard';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('<AnalyticsCard />', () => {
  describe('render()', () => {
    test('renders the AnalyticsCard component', () => {
      const component = shallow(<AnalyticsCard />);
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});