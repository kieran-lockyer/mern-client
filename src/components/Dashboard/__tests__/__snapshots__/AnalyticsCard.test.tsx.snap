// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnalyticsCard /> render() renders the AnalyticsCard component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
